routers/api/v1/admin/user.go

```diff
@@ -182,10 +182,19 @@ func EditUser(ctx *context.APIContext, form api.EditUserOption) {
 	}
 
 	u.LoginName = form.LoginName
-	u.FullName = form.FullName
-	u.Email = form.Email
-	u.Website = form.Website
-	u.Location = form.Location
+
+	if form.FullName != nil {
+		u.FullName = *form.FullName
+	}
+	if form.Email != nil {
```

Comment:
Email is required if `REGISTER_EMAIL_CONFIRM` is true?

Reply:
but if no value is provided than field will not be updated, what this could be is missing is that if empty string is provided it should be checked to return required value error for email

Reply:
@lunny the `REGISTER_EMAIL_CONFIRM` should be no concern - since it is the **admin** interface (you obvious dont have to on the Admin web UI too)